### DISCUSSION TITLE
How to compare every 10 mins of today to yesterday?

### DISCUSSION THREAD

**Discussion by AKA-bingo** [jertel/elastalert2#1568] [Q&A]
I wanna to compare something(such as doc count) every 10 mins of today to yesterday.

when I try to use spike rule type, I found that the time window should be continuous, e.g. [14:00 - 14:10] compared to [13:50 - 14:00]

How can I compare the data from the past 10 minutes with the data from the same time yesterday?

Thank you for you help!

**Comment by jertel** [comment]
As far as I know that's not possible with the Spike rule. It only looks at the previous time window and compares the hit count with the current time window. 

There hasn't been much demand for that type of rule, at least not vocalized on this forum. But if you would like to create a PR to add this support you are welcome to do so, provided the [Contribution Guidlines](../blob/master/CONTRIBUTING.md) are followed. This would be a bit complex to implement and would require a new property that let the admin control how far apart the reference window is from the current window. Additionally, each reference window would actually consist of one or more query_key windows.
